server/main_server/src/main.cpp

```diff
@@ -0,0 +1,1444 @@
+                const auto local_server_port = irods::get_server_property<int>(irods::KW_CFG_ZONE_PORT);
+                const auto local_server_port_string = std::to_string(local_server_port);
+
+                if (is_server_listening_for_connections(local_server_host, local_server_port_string) != 0) {
+                    return;
+                }
+            }
+            catch (const std::exception& e) {
+                log_server::debug("{}: Connect Error: {}. Deferring launch of Delay Server.", __func__, e.what());
+                return;
+            }
+        }
+
+        const auto migration_sleep_time_in_seconds = irods::get_advanced_setting<int>(irods::KW_CFG_MIGRATE_DELAY_SERVER_SLEEP_TIME_IN_SECONDS);
+        const auto now = std::chrono::steady_clock::now();
+
+        if (now - _time_start < std::chrono::seconds{migration_sleep_time_in_seconds}) {
+            return;
+        }
+
+        _time_start = now;
+
+        std::optional<std::string> leader;
+        std::optional<std::string> successor;
+
+        try {
+            irods::experimental::client_connection conn;
+            leader = get_delay_server_leader(conn);
+            successor = get_delay_server_successor(conn);
+
+            if (leader && successor) {
+                log_server::debug("{}: Delay server leader [{}] and successor [{}].", __func__, *leader, *successor);
+
+                // Update the host information based on the "host_resolution" configuration, if defined.
+                leader = get_preferred_host(*leader);
+                successor = get_preferred_host(*successor);
+
+                // 4 cases
+                // ----------------------------------------------------------------------------
+                // C  L  S  R
+                // ----------------------------------------------------------------------------
+                // 0  0  0  Invalid state / No-op
+                // 1  1  0  Delay server migration complete
+                // 2  0  1  Leader has shut down delay server, Successor launching delay server
+                // 3  1  1  Leader running delay server, Admin requested migration
+                // ----------------------------------------------------------------------------
+                // C = Case, L = Leader, S = Successor, R = Result
+
+                // This server is the leader and may be running a delay server.
+                if (local_server_host == *leader) {
+                    // Case 3
+                    if (local_server_host == *successor) {
+                        launch_delay_server(_write_to_stdout, _enable_test_mode);
+
+                        // Clear successor entry in catalog. This isn't necessary, but helps
+                        // keep the admin from becoming confused.
+                        if (g_pid_ds > 0) {
+                            set_delay_server_migration_info(conn, KW_DELAY_SERVER_MIGRATION_IGNORE, "");
```

Comment:
what is this second parameter?

Reply:
Successor host

Reply:
right, but... the keyword was surprising to see.    trust you, but was not clear on reading the source.

Reply:
Wait, I described the third param. The keyword you are asking about is special to the API. It instructs the API to leave the existing value for the leader and/or successor as is. In this call, the leader will remain unchanged.

Reply:
okay, please consider a note - or just resolve this.  no worries.